File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    s = list(s)
    
    result = []
    
    for item in s:
        # 배열이 비어있으면 삽입
        if not result:
            result.append(item)
        else:
            if item == ")" and result[-1] == "(":
                result.pop()
            else:
                result.append(item)
                
    return False if result else True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const stack = [];
    
    for (const item of s) {
        if (!stack.length) stack.push(item);
        else {
            if (stack[stack.length - 1] === "(" && item === ")") {
                stack.pop();
            } else {
                stack.push(item);
            }
        }
    }

    return stack.length ? false : true;
}